[PANIC] Building resource docs is broken

Due to a change introduced in all providers as part of v2.3.3 of the bridge, the schema.json doesn't have the version embedded in it. The version gets passed in before the schema is used in provider generation

https://github.com/pulumi/pulumi-aws/blob/master/Makefile#L34

Notice we are passing `VERSION` are the environment variable. The resource docs are unaware of this change - my apologies for this! - and  I believe this is the cause of a panic:

```
~/code/go/src/github.com/pulumi/docs/tools/resourcedocsgen ~/code/go/src/github.com/pulumi/docs
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17bec9b]

goroutine 1 [running]:
github.com/pulumi/pulumi/pkg/v2/codegen/docs.(*modContext).genIndex(0xc0000bdc00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/stack72/code/go/src/github.com/pulumi/pulumi/pkg/codegen/docs/gen.go:1634 +0x95b
github.com/pulumi/pulumi/pkg/v2/codegen/docs.(*modContext).gen(0xc0000bdc00, 0xc00090b1a0, 0xc0007bf8a0, 0xc00039e7e0)
	/Users/stack72/code/go/src/github.com/pulumi/pulumi/pkg/codegen/docs/gen.go:1474 +0x892
github.com/pulumi/pulumi/pkg/v2/codegen/docs.GeneratePackage(0x19b8702, 0x15, 0xc0000d8480, 0x0, 0x0, 0x0)
	/Users/stack72/code/go/src/github.com/pulumi/pulumi/pkg/codegen/docs/gen.go:1802 +0x3f1
main.generateDocsFromSchema(0x7ffeefbff6c9, 0x56, 0xc000470000, 0x1820000, 0xc000470000)
	/Users/stack72/code/go/src/github.com/pulumi/docs/tools/resourcedocsgen/main.go:189 +0x15e
main.main()
	/Users/stack72/code/go/src/github.com/pulumi/docs/tools/resourcedocsgen/main.go:77 +0x3b1
exit status 2
```

I believe the `gen_resource_docs.sh` needs to change to pass the VERSION parameter to the `go run` command so that it pushes the version back to the schema

The example of how this is handled in the schema can be found in any of the provider generate steps e.g. https://github.com/pulumi/pulumi-aws/blob/master/provider/cmd/pulumi-resource-aws/generate.go

--- a/scripts/gen_resource_docs.sh
+++ b/scripts/gen_resource_docs.sh
@@ -54,11 +54,6 @@ generate_docs() {
         pulumi plugin install resource "${provider}" "${plugin_version}"
     fi
 
-    TFGEN=pulumi-tfgen-${provider}
-    echo "Running pulumi-tfgen-${TFGEN} to generate provider schema..."
-    make generate_schema
-    popd
-
     if [ "$provider" = "kubernetes" ]; then
         SCHEMA_FILE="../../../pulumi-kubernetes/sdk/schema/schema.json"
         OVERLAY_SCHEMA_FILE="./overlays/kubernetes/overlays.json"
@@ -67,12 +62,20 @@ generate_docs() {
         OVERLAY_SCHEMA_FILE=""
     fi
 
-    echo "Removing the ${PACKDIR}/${provider} dir..."
-    rm -rf "${PACKDIR}/${provider}"
+    # Use a previously generated schema.json file if it exists.
+    if [ -f "${SCHEMA_FILE}" ]; then
+        echo "Will use previously generated schema.json for generating docs..."
+    else
+        echo "Could not find a schema.json file. Generating schema..."
+        make generate_schema
+    fi
+
+    # Now leave the provider repo and back to the docs repo.
+    popd
 
     echo "Running docs generator from schema for ${provider}..."
     pushd ${TOOL_RESDOCGEN}
-    go run . -logtostderr "${ABSOLUTEPACKDIR}/${provider}" "${SCHEMA_FILE}" "${OVERLAY_SCHEMA_FILE}" || exit 3
+    go run . -logtostderr "${ABSOLUTEPACKDIR}/${provider}" "${SCHEMA_FILE}" "${plugin_version}" "${OVERLAY_SCHEMA_FILE}" || exit 3
     popd
 
     echo "Done generating resource docs for ${provider}"
@@ -81,8 +84,8 @@ generate_docs() {
 
 if [ -z "${REPO_OVERRIDE:-}" ]; then
     for PROVIDER in "${PROVIDERS[@]}" ; do \
-        generate_docs $PROVIDER
+        generate_docs "$PROVIDER"
     done
 else
-    generate_docs ${REPO_OVERRIDE}
+    generate_docs "${REPO_OVERRIDE}"
 fi

--- a/tools/resourcedocsgen/main.go
+++ b/tools/resourcedocsgen/main.go
@@ -38,14 +38,14 @@ func main() {
 	// Grab the args.
 	flag.Parse()
 	args := flag.Args()
-	if len(args) < 2 {
-		fmt.Fprintf(os.Stderr, "error: usage: %s <out-dir> <provider-schema-file> [overlay-schema-file]>\n", os.Args[0])
+	if len(args) < 3 {
+		fmt.Fprintf(os.Stderr, "error: usage: %s <out-dir> <provider-schema-file> <version> [overlay-schema-file]>\n", os.Args[0])
 		os.Exit(1)
 	}
 
 	defer glog.Flush()
 
-	outDir, schemaFile, overlaysSchemaFile := args[0], args[1], args[2]
+	outDir, schemaFile, version, overlaysSchemaFile := args[0], args[1], args[2], args[3]
 
 	schema, err := ioutil.ReadFile(schemaFile)
 	if err != nil {
@@ -58,6 +58,7 @@ func main() {
 		glog.Infof("error unmarshalling schema into a PackageSpec: %v", err)
 		os.Exit(1)
 	}
+	mainSpec.Version = version
 
 	if overlaysSchemaFile != "" {
 		overlaySpec := &pschema.PackageSpec{}

